rosie/src/main/java/com/karumi/rosie/view/PresenterLifeCycleLinker.java

```diff
@@ -78,13 +87,63 @@ public void pausePresenters() {
     }
   }
 
+  /**
+   * Destroys all the already registered presenters lifecycle.
+   */
   public void destroyPresenters() {
     for (RosiePresenter presenter : presenters) {
       presenter.destroy();
     }
   }
 
+  /**
+   * Registers a presenter instance.
+   *
+   * @param presenter to be registered
+   */
   public void registerPresenter(RosiePresenter presenter) {
+    if (presenter == null) {
+      throw new IllegalArgumentException("The presenter instance to be registered can't be null");
+    }
     presenters.add(presenter);
   }
+
+  /**
+   * Updates the view instance associated to all the already registered presenters.
+   *
+   * @param view to be assigned to the presenters.
+   */
+  public void setView(RosiePresenter.View view) {
```

Comment:
We can register a presenter manually and that's why the visibility of this method has to be public. As we can register a presenter which view has not been initialized we need to keep this method to update the view instance if needed. On the other hand, imagine that the framework user wants to change the view without invoke any lifecycle method, we need to keep this method as public.